src/Plug.vala

```diff
@@ -54,36 +54,33 @@ namespace SwitchboardPlugUserAccounts {
 
             infobar_error = new Gtk.InfoBar ();
             infobar_error.message_type = Gtk.MessageType.ERROR;
-            infobar_error.no_show_all = true;
+            infobar_error.revealed = false;
 
             var error_label = new Gtk.Label ("");
 
             var error_content = infobar_error.get_content_area ();
             error_content.add (error_label);
 
-            InfobarNotifier.get_default ().error_notified.connect (() => {
-                if (InfobarNotifier.get_default ().is_error ()) {
-                    infobar_error.no_show_all = false;
-                    error_label.label = "%s: %s".printf (_("Password change failed"), InfobarNotifier.get_default ().get_error_message ());
+            InfobarNotifier.get_default ().notify["error-message"].connect (() => {
+                var error_message = InfobarNotifier.get_default ().error_message;
+                if (error_message != "") {
+                    error_label.label = "%s: %s".printf (_("Password change failed"), error_message);
                     infobar_error.show_all ();
```

Comment:
I guess the `infobar_error` should be displayed without `show_all`; just changing the value of `revealed` works.